scripts/compress-outputs.sh

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env bash
+
+# This will be chatty, so we don't use `set -x`
+set -eu # Errors/undefined vars are fatal
+set -o pipefail # Check all commands in a pipeline
+
+# Compress all of our HTML and raw-analysis files which end up being very large
+# and only exist to be served via nginx and therefore can be pre-compressed,
+# resulting in a net win for everyone.
+#
+# Limitations in our use of try_files by nginx and the `gzip_static` logic mean
+# that:
+# - In cases where we're using try_files we need to have the normal
+#   non-gz-suffixed file on disk for try_files to be able to identify what
+#   directory the file is in from its search list.  This need not be the
+#   original file, so we can be tricky and just create a zero-length version of
+#   the file because no one will read it.
+# - The gzipped versions of the files need to just have the straightforward .gz
+#   suffix.
+#
+# If you were thinking "What if the tree actually contains both FOO and FOO.gz?"
+# then you win a prize.  And that prize is either coming up with an elegant fix
+# to the problem or being party to the inelegant fix we depend on.
+#
+# It turns out mozilla-central has exactly this case in the guise of
+# `devtools/client/styleeditor/test/simple.css` and its gzipped twin
+# `devtools/client/styleeditor/test/simple.css.gz`.  Right now (pre this patch)
+# if you view the ".gz" file you'll see a useless single gibberish line of text.
+#
+# Our fix:
+# - We observe that if we process the existing "FOO.gz" file before "FOO", then
+#   when FOO's gzipping overwrites "FOO.gz", there's no harm because it's
+#   overwriting a zero-length file that only needed to exist so its filename
```

Comment:
Technically FOO.gz is not going to be a zero-length file. It will contain the original FOO.gz contents (which got compressed into FOO.gz.gz) with the timestamp updated via touch to match that of FOO.gz.gz. But your point remains valid that FOO.gz is only used to satisfy the try_files requirement and not actually read when the user requests FOO.gz, so it's ok that the file contains the compressed version of FOO instead.

Reply:
I think you're describing what happens if we do `gzip -k` for keep.  If we have a file `FOO` and run `gzip FOO`, then afterwards we only have a file `FOO.gz`.  The zero length file is created by using `touch`.

Reply:
Ah yes, you're right, my mistake!